src/lib/metrics.test.ts

```diff
@@ -93,7 +99,32 @@ test('should collect metrics for updated toggles', async () => {
 
     const metrics = await prometheusRegister.metrics();
     expect(metrics).toMatch(
-        /feature_toggle_update_total\{toggle="TestToggle",project="default",environment="default"\} 1/,
+        /feature_toggle_update_total\{toggle="TestToggle",project="default",environment="default",environmentType="n\/a"\} 1/,
+    );
+});
+
+test('should set environmentType when toggle is flipped', async () => {
+    await environmentStore.create({
+        name: 'testEnvironment',
+        enabled: true,
+        type: 'testType',
+        sortOrder: 1,
+    });
+    stores.eventStore.emit(FEATURE_ENVIRONMENT_ENABLED, {
+        featureName: 'TestToggle',
+        project: 'default',
+        environment: 'testEnvironment',
+        data: { name: 'TestToggle' },
+    });
+
+    // Wait for event to be processed, not nice, but it works.
+    await new Promise((done) => {
+        setTimeout(done, 10);
```

Comment:
does it also work with setTimeout(done, 1)?

Reply:
also you can search for `await ms(100);` this pattern in our codebase that better reveals the intent

Reply:
fixed

Reply:
ops was wrong i have not fixed

Reply:
on my machine 1ms is enough